cobra/flux_analysis/sampling.py

```diff
@@ -637,14 +681,17 @@ def sample(self, n, fluxes=True):
             n = n_process * self.np
             # No with statement or starmap here since Python 2.x
             # does not support it :(
-            mp = Pool(self.np)
+            mp = Pool(self.np, initializer=mp_init, initargs=(self,))
             chains = mp.map(
                 _sample_chain,
-                zip([self] * self.np, [n_process] * self.np, range(self.np)),
+                zip([n_process] * self.np, range(self.np)),
                 chunksize=1)
```

Comment:
So, no pool in the process is ever recycled. A pool of size n will receive exactly n problems to sample, so imap_unordered does not really make sense IMO. A chunksize of one guarantees that each process will run one sampling chain otherwise it would *theoretically* be possible that a process received two chains and another none (unlikely).